charts/k8ssandra/templates/stargate/stargate.yaml

```diff
@@ -77,16 +77,16 @@ spec:
           resources:
```

Comment:
I think it makes sense to make `resources` in general configurable. See [here](https://github.com/k8ssandra/k8ssandra/blob/main/charts/k8ssandra/templates/cassandra/cassdc.yaml#L53) for an example. 

Given that we have some insights into memory configuration I would be inclined to do as follows:

* If no values are set the apply defaults as you currently do
* else if heap/memory is specified, use those overrides
* else if `resources` is specified, use that.

The user directly setting `resources` would be a more advanced scenario IMO but one that we definitely want to support.